rlbot_gui/gui/js/main.js

```diff
@@ -7,6 +7,9 @@ import Main from './main-vue.js'
 import Sandbox from './sandbox-vue.js'
 import Story from './story-mode.js'
 
+// eel does not provide an API for this. Close the browser when the websocket closes.
+document.addEventListener("DOMContentLoaded", e => eel._websocket.onclose = window.close.bind(window));
```

Comment:
Is eel._websocket guaranteed to be defined when we get the DOMContentLoaded event? I'm worried it might not be, is there any other event or callback we could hook into?

Reply:
It is guaranteed to be defined already since we add the event listener after eel does this, it will always run after eels event where it defines this websocket. (If 2 event listeners bind to the same event, by spec the one that was bound first will run first)

Our main.js always runs after eels code runs and defines this event listener, if main.js somehow ran before eels code, main.js would never even get to this line and already have crashed at line 1.

Reply:
Ok cool. Thanks for this change!